src/wagtailmedia/settings.py

```diff
@@ -0,0 +1,106 @@
+"""
+The wagtailmedia settings are namespaced in the WAGTAILMEDIA setting.
+For example your project's `settings.py` file might look like this:
+WAGTAILMEDIA = {
+    "MEDIA_MODEL": "mymedia.CustomMedia",
+    # ...
+}
+This module provides the `wagtailmedia_settings` object, that is used to access
+the settings. It checks for user settings first, with fallback to defaults.
+"""
+import warnings
+
+from django.conf import settings
+from django.test.signals import setting_changed
+
+
+DEFAULTS = {"MEDIA_MODEL": "wagtailmedia.Media", "MEDIA_FORM_BASE": ""}
+
+# List of settings that have been deprecated
+DEPRECATED_SETTINGS = [
+    "WAGTAILMEDIA_MEDIA_MODEL",
+    "WAGTAILMEDIA_MEDIA_FORM_BASE",
+]
+
+# List of settings that have been removed
+REMOVED_SETTINGS = []
+
+
+class WagtailMediaSettings:
+    """
+    A settings object that allows the wagtailmedia settings to be accessed as
+    properties. For example:
+        from wagtailmedia.settings import wagtailmedia_settings
+        print(wagtailmedia_settings.MEDIA_MODEL)
+    Note:
+    This is an internal class that is only compatible with settings namespaced
+    under the WAGTAILMEDIA name. It is not intended to be used by 3rd-party
+    apps, and test helpers like `override_settings` may not work as expected.
+    """
+
+    def __init__(self, user_settings=None, defaults=None):
+        if user_settings:
+            self._user_settings = self.__check_user_settings(user_settings)
+        self.defaults = defaults or DEFAULTS
+        self._cached_attrs = set()
+
+    @property
+    def user_settings(self):
+        if not hasattr(self, "_user_settings"):
+            self._user_settings = self.__check_user_settings(
+                getattr(settings, "WAGTAILMEDIA", {})
+            )
+        return self._user_settings
+
+    def __getattr__(self, attr):
+        if attr not in self.defaults:
+            raise AttributeError("Invalid wagtailmedia setting: '%s'" % attr)
+
+        try:
+            # Check if present in user settings
+            val = self.user_settings[attr]
+        except KeyError:
+            # Fall back to defaults
+            val = self.defaults[attr]
+
+        # Cache the result
+        self._cached_attrs.add(attr)
+        setattr(self, attr, val)
+        return val
+
+    def __check_user_settings(self, user_settings):
+        for setting in DEPRECATED_SETTINGS:
+            if setting in user_settings or hasattr(settings, setting):
+                new_setting = setting.replace("WAGTAILMEDIA_", "")
+                warnings.warn(
+                    f"The '{setting}' setting is deprecated and will be removed in the next release, "
+                    f'use WAGTAILMEDIA["{new_setting}"] instead.',
+                    category=PendingDeprecationWarning,
```

Comment:
Might be a good idea to create some ``RemovedInWagtailMediaVersionX`` sub-classes to help track when settings need to be moved to ``DeprecationWarning`` and then removed

Reply:
Good suggestion! Thought of doing that initially, but then was too lazy 😂